conn.go

```diff
@@ -270,6 +284,44 @@ func DialWithContext(ctx context.Context, network string, raddr *net.UDPAddr, co
 	return ClientWithContext(ctx, pConn, config)
 }
 
+type connPacket struct {
```

Comment:
mind moving into its own file?

As a reader it is to just ignore a bunch of stubs. As I move through a large file I find it overwhelming otherwise.